docs/product-updates/2023-08-01-custom-code-samples.md

```diff
@@ -0,0 +1,12 @@
+---
+title: Custom code samples
+tags: [Improvement]
+---
+
+You would like to make your API consumers feel more at home by providing them some in-house custom code sample to use your API?
```

Comment:
```suggestion
The OpenAPI specification [supports setting custom example values](https://spec.openapis.org/oas/v3.1.0#example-object) for a request or a response. It makes it possible for Bump.sh to display those examples, as the third column of our UI layout.

By the way, if you haven't set any example, Bump.sh will generate some automatically and still populate your doc for an improved developer experience.

Now those examples are generated as a cURL request, which is convenient to copy-paste into a terminal for testing purposes, but isn't necessarily how developers consuming your API will interact with it on a day-to-day basis.

Bump.sh now supports the `x-codeSamples` extension, which lets you set custom code samples in any programming language you expect your users to work with.

Add any language and related examples, and doc viewers will be able to select it directly from a drop-down.
```

Reply:
can I make a suggestion on Chris' suggestion ?

```
The OpenAPI specification [supports setting custom example values](https://spec.openapis.org/oas/v3.1.0#example-object) for a request or a response. It makes it possible for Bump.sh to display those examples, as the third column of our UI layout.
By default, if you haven't set any examples, Bump.sh will generate them automatically and populate your doc for an improved developer experience.
These examples are generated as a cURL request, which is convenient to copy-paste into a terminal for testing purposes, but isn't necessarily how developers consuming your API will interact with it on a day-to-day basis.

Bump.sh now supports the `x-codeSamples` extension, which lets you set custom code samples in any programming language you expect your users to work with.
Add code samples in as much languages you'd like, and your documentation readers will be able to select their prefered language from a drop down menu.
```

Reply:
Also a screenshot of the dropdown menu in usage would be a nice addition here

Reply:
I like the suggestion, however beware example objects and code sample objects are two completely independent things:
- OpenAPI [Example object](https://spec.openapis.org/oas/v3.1.0#example-object)s are a way to display custom **request** or **response** examples (usually a JSON document)
- The added code samples have strictly no relation with the Example objects mentioned above. There goal is to provide a code sample on how to use the API in a specific programming language

By reading the suggestion from Chris I feel there can be a confusion (not to mention that this current improvement doesn't change any of the behavior of `examples` displayed in our generated API documentations).

Was it on purpose to mention the “Example object”s from the OpenAPI in this product update even if we didn't improve anything around those here?

Reply:
Don't you feel like we are giving too much detail here, for intance what about keeping that kind of infos for the help center:

> By default, if you haven't set any examples, Bump.sh will generate them automatically and populate your doc for an improved developer experience.
These examples are generated as a cURL request, which is convenient to copy-paste into a terminal for testing purposes, but isn't necessarily how developers consuming your API will interact with it on a day-to-day basis.

And having the update centered around the benefice and value of the support?

Reply:
And prioritize the explaination at the beginning of the entry?

> Bump.sh now supports the `x-codeSamples` extension, which lets you set custom code samples in any programming language you expect your users to work with.
Add code samples in as much languages you'd like, and your documentation readers will be able to select their prefered language from a drop down menu.

Reply:
> Add code samples in as much languages you'd like, and your documentation readers will be able to select their prefered language from a drop down menu.

That's powerful and need to be read asap for the customer I feel 👼